SDWebImage/SDImageCache.m

```diff
@@ -248,7 +248,8 @@ - (NSData *)diskImageDataBySearchingAllPathsForKey:(NSString *)key {
         return data;
     }
 
-    for (NSString *path in self.customPaths) {
+    NSArray *customPaths = [self.customPaths copy];
```

Comment:
from one pov is a workaround, but this solution fixes the issue, plus it assures we don't get into situations like this in the future, if other non-synchronized code is added. Synchronising the access to this resources would have a performance impact